sparkly/reader.py

```diff
@@ -214,6 +219,77 @@ def mysql(self, host, database, table, port=None, parallelism=None, options=None
 
         return self._basic_read(reader_options, options, parallelism)
 
+    def kafka(self,
+              host,
+              topic,
+              offset_ranges=None,
+              key_deserializer=None,
+              value_deserializer=None,
+              schema=None,
+              port=None,
+              parallelism=None,
+              options=None):
+        """Creates dataframe from specified set of messages from Kafka topic.
+
+        Defining ranges:
+            - If `offset_ranges` is specified it defines which specific range to read.
+            - If `offset_ranges` is omitted it will auto-discover it's partitions.
+
+        If `schema` parameter is specified the Dataframe with this schema will be returned,
+        else the RDD will be returned.
+        The `schema` parameter, if specified, should contain two top level fields:
+        `key` and `value`.
+
+        Args:
+            host (str): Kafka host.
+            topic (str|None): Kafka topic to read from.
+            offset_ranges (list[(int, int, int)]|None): List of partition ranges
+                [(partition, start_offset, end_offset)].
+            key_deserializer (function): Function used to deserialize the key.
+            value_deserializer (function): Function used to deserialize the value.
+            schema (pyspark.sql.types.StructType|None): Schema to apply to create a Dataframe.
+            port (int|None): Kafka port.
+            parallelism (int|None): The max number of parallel tasks that could be executed
+                during the read stage (see :ref:`controlling-the-load`).
+            options (dict|None): Additional kafka parameters, see KafkaUtils.createRDD docs.
+
+        Returns:
+            pyspark.sql.DataFrame
+
+        Raises:
+            InvalidArgumentError
+        """
+        assert self._hc.has_package('org.apache.spark:spark-streaming-kafka')
+
+        if not key_deserializer or not value_deserializer or not schema:
```

Comment:
- If prefer grouping `topic` with `offset_ranges` (which is optional)
- I don't like logic in assertions, as it could be suppressed with -O flag in py2
- It's possible that in future this arguments will be optional